src/range.rs

```diff
@@ -0,0 +1,82 @@
+use core::ops::{ RangeBounds, Bound };
+use core::iter::FusedIterator;
+use crate::Itertools;
+
+/// An iterator over a range of values.
+///
+/// Acquired by the [`range`] function or the
+/// [`Itertools::range`] method.
+pub struct Range<I, R> {
+    range: R,
+    internal: I,
+    counter: usize,
+}
+
+impl<I, R> Clone for Range<I, R> 
+    where I: Clone, R: Clone 
+{
+    fn clone(&self) -> Self {
+        Range {
+            counter: self.counter,
+            range: self.range.clone(),
+            internal: self.internal.clone(),
+        }
+    }
```

Comment:
We canonically use `clone_fields` to avoid boilerplate. You could also add a `Debug` impl with `debug_fmt_fields`.